.github/workflows/temp-check-legacy-links-format.yml

```diff
@@ -0,0 +1,126 @@
+
+on:
+  workflow_call:
+    inputs:
+      repo-owner:
+        default: 'hashicorp'
+        required: false
+        type: string
+      repo-name:
+        required: true
+        type: string
+      commit-sha:
+        required: true
+        type: string
+
+jobs:
+  docs-check-legacy-links-format:
+    runs-on: ubuntu-latest
+    env:
+      OWNER: ${{ inputs.repo-owner }}
+      REPO: ${{ inputs.repo-name }}
+      COMMIT_SHA: ${{ inputs.commit-sha }}
+    steps:
+      - name: Get PR number
+        id: get-pr-number
+        uses: actions/github-script@v6
+        with:
+          script: |
+            // TODO don't hardcode owner, repo, commit_sha via process.env
+            const { data } = await github.rest.repos.listPullRequestsAssociatedWithCommit({
+              owner: process.env.OWNER,
+              repo: process.env.REPO,
+              commit_sha: process.env.COMMIT_SHA
+            })
+            return data[0].number
+
+      - name: Get list of PR files
+        id: get-lists-of-pr-files
+        uses: actions/github-script@v6
+        env:
+          PR_NUMBER: ${{ steps.get-pr-number.outputs.result }}
+        with:
+          script: |
+            const changedMdxFiles = []
+            const changedNavDataJsonFiles = []
+
+            // A helper for recursively fetching all files for a PR
+            const fetchAllPrFiles = async ({ allPrFiles = [], page = 1 }) => {
+              const { data } = await github.rest.pulls.listFiles({
+                owner: process.env.OWNER,
+                repo: process.env.REPO,
+                per_page: 100,
+                pull_number: process.env.PR_NUMBER,
+                page,
+              })
+              data.forEach(file => {
+                allPrFiles.push(file)
+              })
+
+              if (data.length === 100) {
+                await fetchAllPrFiles({ allPrFiles, page: page + 1 })
+              }
+            }
+
+            // Fetch all the PR files
+            const allPrFiles = []
+            await fetchAllPrFiles({ allPrFiles })
+
+            // Sort the PR files for what we're interested in
+            allPrFiles.forEach(file => {
+              if (file.status === 'removed') {
+                return
+              }
+              if (file.filename.endsWith('.mdx')) {
+                changedMdxFiles.push(file.filename)
+                return
+              }
+              if (file.filename.endsWith('-nav-data.json')) {
+                changedNavDataJsonFiles.push(file.filename)
+              }
+            })
+
+            // Log and return the result
+            const result = { changedMdxFiles, changedNavDataJsonFiles }
+            console.log('get-lists-of-pr-files result:', result)
+            return result
+
+      - name: Checkout dev-portal
+        uses: actions/checkout@v3
+        with:
+          repository: 'hashicorp/dev-portal'
+          ref: 'amb.explore-docs-content-links'
```

Comment:
TODO remove this before merging, was only present for testing in https://github.com/hashicorp/waypoint/pull/4277

Reply:
Resolving because the comment became outdated. Made a new one!